nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/GetLastDocumentVersionTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Ricardo Dias
+ */
+
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import static org.junit.Assert.*;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.*;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.VersioningOption;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.versioning.VersioningService;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import com.google.inject.Inject;
+
+/**
+ * @author rdias
+ * @since 8.3
+ */
+
+@RunWith(FeaturesRunner.class)
+@Features({CoreFeature.class})
+@Deploy("org.nuxeo.ecm.automation.core")
+public class GetLastDocumentVersionTest {
+
+    @Inject
+    CoreSession session;
+
+    @Inject
+    AutomationService service;
+
+    protected DocumentModel folder;
+
+    protected DocumentModel section;
+
+    protected DocumentModel docWithMajorVersions;
+
+    protected DocumentModel docWithMinorVersions;
+
+    protected DocumentModel docWithMajorMinorVersions;
+
+    protected DocumentModel docWithNoVersion;
+
+    @Before
+    public void initRepo() throws Exception {
+        session.removeChildren(session.getRootDocument().getRef());
+        session.save();
+
+        folder = session.createDocumentModel("/", "Folder", "Folder");
+        folder.setPropertyValue("dc:title", "Folder");
+        folder = session.createDocument(folder);
+        session.save();
+        folder = session.getDocument(folder.getRef());
+
+        // major versions
```

Comment:
No need to create all documents at setup given only one is used in each test.
Better to have a single DocumentModel field and initialise it at the beginning of each test, easier to understand.